### ISSUE TITLE
The "RaiseAutomationEvent" method doesn't check "IsHandleCreated" flag.

### ISSUE THREAD

**Issue by SergeySmirnov-Akvelon** [dotnet/winforms#3419]
- .NET Core Version:
  .NET 5 P4 

**Problem description:**

https://github.com/dotnet/winforms/pull/3228#discussion_r437081185

Use of `RaiseAutomationEvent`, `RaiseAutomationPropertyChangedEvent`, etc. requires guards again a non-existing handle here, else we'll force a creation. However it is easy to overlook this fact, so the guard `if (IsHandleCreated)` must be implemented in those events rather than expect the caller to apply the guard.

E.g. `AccessibilityNotifyClients` have the guard:
https://github.com/dotnet/winforms/blob/d4cfc2d9d6cb20984761fc0cf5eb8efd70c2d810/src/System.Windows.Forms/src/System/Windows/Forms/Control.cs#L4505-L4511

`RaiseAutomation*Event` must perform the same checks, but since they are defined on `AccessibleObject` and have no notion of `owner` object, perhaps these should be moved to the level of `ControlAccessibleObject`, or `ControlAccessibleObject` should provide an override.

**Expected behavior:**

`RaiseAutomation*Event` method should check `IsHandleCreated` flag.
